packages/core/src/connectors/alipay/index.ts

```diff
@@ -0,0 +1,237 @@
+      re_expires_in: string; // Expiring time of refresh token, in seconds
+    };
+  };
+
+  const config = await getConnectorConfig<AlipayConfig>(metadata.id);
+  const initSearchParameters = {
+    method: methodForAccessToken,
+    format: 'JSON',
+    timestamp: getFormattedDate(),
+    version: '1.0',
+    grant_type: 'authorization_code',
+    code: authCode,
+    charset: 'UTF8',
+    ...config,
+  };
+  const signedSearchParameters = signingPamameters(initSearchParameters);
+
+  const response = await got
+    .post(alipayEndpoint, {
+      searchParams: signedSearchParameters,
+      timeout: await getConnectorRequestTimeout(),
+    })
+    .json<AccessTokenResponse>();
+
+  const { msg, sub_msg } = response.error_response ?? {};
+  assertThat(
+    response.alipay_system_oauth_token_response,
+    new ConnectorError(
+      ConnectorErrorCodes.SocialAuthCodeInvalid,
+      msg ? msg : sub_msg ? sub_msg : undefined
+    )
+  );
+  const { access_token: accessToken } = response.alipay_system_oauth_token_response;
+
+  assertThat(accessToken, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+
+  return { accessToken };
+};
+
+export const getUserInfo: GetUserInfo = async (accessTokenObject) => {
+  type UserInfoResponse = {
+    sign: string; // To know `sign` details, see: https://opendocs.alipay.com/common/02kf5q
+    alipay_user_info_share_response: {
+      user_id?: string; // String of digits with max length of 16
+      avatar?: string; // URL of avatar
+      province?: string;
+      city?: string;
+      nick_name?: string;
+      gender?: string; // Enum type: 'F' for female, 'M' for male
+      code?: string;
+      msg?: string; // To know `code` and `msg` details, see: https://opendocs.alipay.com/common/02km9f
+      sub_code?: string;
+      sub_msg?: string;
+    };
+  };
+
+  const { accessToken } = accessTokenObject;
+
+  const config = await getConnectorConfig<AlipayConfig>(metadata.id);
+  const initSearchParameters = {
+    method: methodForUserInfo,
+    format: 'JSON',
+    timestamp: getFormattedDate(),
+    version: '1.0',
+    grant_type: 'authorization_code',
+    auth_token: accessToken,
+    biz_content: JSON.stringify({}),
+    charset: 'UTF8',
+    ...config,
+  };
+  const signedSearchParameters = signingPamameters(initSearchParameters);
+
+  const response = await got
+    .post(alipayEndpoint, {
+      searchParams: signedSearchParameters,
+      timeout: await getConnectorRequestTimeout(),
+    })
+    .json<UserInfoResponse>();
+
+  const {
+    user_id: id,
+    avatar,
+    nick_name: name,
+    sub_msg,
+    sub_code,
+    msg,
+    code,
+  } = response.alipay_user_info_share_response;
+
+  assertThat(
+    !msg && !sub_msg && !code && !sub_code,
+    new ConnectorError(
+      ConnectorErrorCodes.SocialAccessTokenInvalid,
+      msg ? msg : sub_msg ? sub_msg : undefined
+    )
+  );
+
+  assertThat(id, new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid));
```

Comment:
Is not access token invalid, may be throw another error, something like invalid response.

Reply:
Fixed.